go/service/teams.go

```diff
@@ -235,17 +237,31 @@ func (h *TeamsHandler) TeamRename(ctx context.Context, arg keybase1.TeamRenameAr
 	return teams.RenameSubteam(ctx, h.G().ExternalG(), arg.PrevName, arg.NewName)
 }
 
+var tokenRegexp = regexp.MustCompile(`token\: [a-z0-9+]{16,17}`)
+
+func parseInviteToken(token string) string {
+	// If the person pasted the whole seitan SMS message in, then let's parse out the token
+	if strings.Contains(token, "token: ") {
+		m := tokenRegexp.FindStringSubmatch(token)
+		if len(m) == 1 {
+			return strings.Split(m[0], " ")[1]
+		}
+	}
+	return token
+}
+
 func (h *TeamsHandler) TeamAcceptInvite(ctx context.Context, arg keybase1.TeamAcceptInviteArg) (err error) {
 	ctx = libkb.WithLogTag(ctx, "TM")
 	defer h.G().CTraceTimed(ctx, "TeamAcceptInvite", func() error { return err })()
 	if err := h.assertLoggedIn(ctx); err != nil {
 		return err
 	}
-	seitan, err := teams.GenerateIKeyFromString(arg.Token)
+	token := parseInviteToken(arg.Token)
```

Comment:
Is it the right function to change? GUI calls `TeamAcceptInviteOrRequestAccess` from this screen:

**Join a team**

*Token or team name*

is there another screen that user is supposed to be pasting messages to that would be calling `TeamAcceptInvite`?

Reply:
There doesn't seem to be any harm in changing both, there is nothing stopping them from calling this one too at some point.